File: go/bob/bob.go
package bob

import "strings"

const testVersion = 2

func Hey(message string) string {
	message = strings.TrimSpace(message)

	if len(message) == 0 {
		return "Fine. Be that way!"
	}

	if isShouting(message) {
		return "Whoa, chill out!"
	}

	if isQuestion(message) {
		return "Sure."
	}

	return "Whatever."
}

func isQuestion(message string) bool {
	return strings.HasSuffix(message, "?")
}

func isShouting(message string) bool {
	return message == strings.ToUpper(message) && message != strings.ToLower(message)
}

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    input =  String.trim(input)

    cond do
      shout?(input) && ask?(input) -> "Calm down, I know what I'm doing!"
      ask?(input) -> "Sure."
      silence?(input) -> "Fine. Be that way!"
      shout?(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end

  defp shout?(input) do
    String.upcase(input) == input && String.downcase(input) != input
  end

  defp ask?(input) do
    String.ends_with?(input, "?")
  end

  defp silence?(input) do
    input == ""
  end
end
